events/events.go

```diff
@@ -171,7 +171,11 @@ type GrenadeEventIf interface {
 // GrenadeEvent contains the common attributes of nade events. Dont register
 // handlers on this tho, you want GrenadeEventIf for that
 type GrenadeEvent struct {
+	// Deprecated: GrenadeType exists for historical compatibility
+	// and should not be used. To access the weapon type corresponding to his GrenadeEvent,
+	// use the Grenade.Weapon instead.
```

Comment:
remove deprecation, `Grenade` may be nil for `InfernoStart` & `InfernoExpired` since we don't know which player threw an inferno (at least in old demos, not sure about new ones)